app/src/main/java/com/github/multimatum_team/multimatum/DeadlineDetailsActivity.kt

```diff
@@ -31,22 +29,131 @@ class DeadlineDetailsActivity : AppCompatActivity() {
 
         // Set the texts for the title and the date of the deadline
-        findViewById<TextView>(R.id.deadline_details_activity_title).text = title
-        findViewById<TextView>(R.id.deadline_details_activity_date).text =
-            getString(R.string.DueTheX, date)
+        titleView.text = SpannableStringBuilder(title)
+        dateView.text = getString(R.string.DueTheXatX, date.toLocalDate(), date.toLocalTime())
+
+        // Set the View to be unmodifiable at the start and remove displacement of the texts
+        fixArrangement()
+
+        // Setup the CheckBox to be checked if done
+        doneButton.isChecked = (state == DeadlineState.DONE)
+        doneButton.setOnCheckedChangeListener { _, isChecked ->
+            state = if (isChecked) DeadlineState.DONE else DeadlineState.TODO
+            updateDetail()
+        }
 
         // Set the detail text to inform the user if it is due, done or the remaining time
-        val detailView = findViewById<TextView>(R.id.deadline_details_activity_done_or_due)
+        updateDetail()
+
+    }
+
+    // This display a DatePickerDialog and afterward a TimePickerDialog to modify the date
+    fun changeDateAndTime(view: View) {
+        val dateSetListener =
+            DatePickerDialog.OnDateSetListener { _, year, monthOfYear, dayOfMonth ->
+                date = LocalDateTime.of(
+                    // monthOfYear is based on an enum that begin with 0
+                    // thus we need to increase it by one to have the true date
+                    year, monthOfYear + 1, dayOfMonth,
+                    date.hour, date.minute
+                )
+                selectTime()
+            }
+
+        // Prepare the datePickerDialog
+        val datePickerDialog = DatePickerDialog(
+            this, dateSetListener,
+            date.year, date.month.ordinal, date.dayOfMonth
+        )
+        // Show the Dialog on the screen
+        datePickerDialog.show()
+    }
+
+    // Function that put the activity to the Edit Mode or to the Uneditable Mode
+    fun goToModifyOrBack(view: View) {
+        if (editMode) {
+            editTitle(true)
+            findViewById<ImageButton>(R.id.deadline_details_activity_modify)
+                .setImageResource(android.R.drawable.checkbox_on_background)
+
+            dateView.setBackgroundResource(android.R.drawable.btn_default)
+            dateView.isClickable = true
+
+            doneButton.isClickable = true
+            doneButton.visibility = View.VISIBLE
+
+        } else {
+            editTitle(false)
+            findViewById<ImageButton>(R.id.deadline_details_activity_modify)
+                .setImageResource(android.R.drawable.ic_menu_manage)
+
+            dateView.isClickable = false
+            dateView.setBackgroundResource(android.R.color.transparent)
+
+            doneButton.isClickable = false
+            doneButton.visibility = View.GONE
+            //TODO: modify the deadline in the repository
+
+        }
```

Comment:
The `if` and the `else` branches look very similar, I think you can refactor the function to make it more concise and avoid code duplication. E.g.:
```Scala
fun goToModifyOrBack(view: View) {
   editTitle(editMode)
   findViewById<ImageButton>(R.id.deadline_details_activity_modify)
                .setImageResource(
                    if (editMode) android.R.drawable.checkbox_on_background
                    else android.R.drawable.ic_menu_manage
                )
   ...
}
```
(Extracting a function might work as well.)